lib/node_modules/@stdlib/stats/base/smeanli/src/main.c

```diff
@@ -27,24 +28,31 @@
 *
 * -   Ling, Robert F. 1974. "Comparison of Several Algorithms for Computing Sample Means and Variances." _Journal of the American Statistical Association_ 69 (348). American Statistical Association, Taylor & Francis, Ltd.: 859–66. doi:[10.2307/2286154](https://doi.org/10.2307/2286154).
 *
-* @param N       number of indexed elements
-* @param X       input array
-* @param stride  stride length
-* @return        output value
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @return         output value
 */
-float stdlib_strided_smeanli( const int64_t N, const float *X, const int64_t stride ) {
-	int64_t ix;
+float API_SUFFIX(stdlib_strided_smeanli)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
+	const CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX(stdlib_strided_smeanli_ndarray)( N, X, strideX, ox );
+}
 
+/**
+* Computes the arithmetic mean of a single-precision floating-point strided array using a one-pass trial mean algorithm and alternative indexing semantics.
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @param offsetX  starting index
+* @return         output value
+*/
+float API_SUFFIX(stdlib_strided_smeanli_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
 	if ( N <= 0 ) {
 		return 0.0f / 0.0f; // NaN
 	}
-	if ( N == 1 || stride == 0 ) {
-		return X[ 0 ];
-	}
-	if ( stride < 0 ) {
-		ix = (1-N) * stride;
-	} else {
-		ix = 0;
+	if ( N == 1 || strideX == 0 ) {
+		return X[ offsetX ];
 	}
-	return X[ ix ] + (float)( (double)stdlib_strided_sapxsum( N-1, -X[ ix ], X+( (stride > 0) ? stride : 0 ), stride ) / (double)N );
+	return X[ offsetX ] + ( API_SUFFIX(stdlib_strided_sapxsum_ndarray)( N-1, -X[ offsetX ], X, strideX, offsetX + strideX ) / (float)N );
```

Comment:
```suggestion
	return X[ offsetX ] + (float)( API_SUFFIX(stdlib_strided_sapxsum_ndarray)( N-1, -X[ offsetX ], X, strideX, offsetX + strideX ) / (double)N );
```

The previous type casts were intentional. We need to compute the mean in double-precision so as not to limit the number of indexed elements `N`. The max safe single-precision float is muuuuuch smaller than the max safe double-precision. If we cast to `float`, we run of the risk of losing precision in the output result, something which the trial-pass mean algorithm is explicitly intended to avoid.

Reply:
i'll take care of that from now on!